.github/workflows/pr-benchmarks.yml

```diff
@@ -0,0 +1,77 @@
+# Benchmark tests for pull requests - runs only if initial and full tests pass
+name: Benchmarks
+
+on:
+  workflow_run:
+    workflows: ["Initial Tests", "Full Tests"]
```

Comment:
Benchmarks were intentionally executed after the initial test passed without waiting for the full test to pass.